src/setupProxy.js

```diff
@@ -24,14 +24,27 @@ function setupTestProxy(app) {
 
   app.use((req, res, next) => {
     const parts = url.parse(req.url);
-    const integrateQuery = parts.pathname + (parts.search ? encodeURIComponent(parts.search) : '');
-    const filePath = path.join(__dirname, 'test/fixtures', integrateQuery);
-
-    if (fs.existsSync(filePath)) {
-      fs.readFile(filePath, 'utf8', (err, contents) => {
-        res.setHeader('Content-Type', 'application/json');
-        res.send(contents);
-      });
+    const filePath = path.join(__dirname, 'test/fixtures', parts.pathname);
+    const queryFilePath = path.join(filePath, parts.search ? encodeURIComponent(parts.search) : '');
+    const indexFilePath = path.join(filePath, 'index.html');
+
+    const sendFile = path => fs.readFile(path, (err, contents) => {
+      res.end(contents);
+    });
+
+    const isFile = path =>
+      fs.existsSync(path)
+      && fs.existsSync(fs.realpathSync(path))
```

Comment:
Why is `realPathSync` needed here? And why is it not necessary for checking below in `isDirectory`?

Reply:
because the fixtures use symlinks to allow loading the archive files with either short or uid, so the second check makes sure the symlink is valid. the fixtures don't use any symlinks on directories so i didn't bother.